### ISSUE TITLE
Request: Inject footer to Posts via post.hbs

### ISSUE THREAD

**Issue by alexellis** [TryGhost/Ghost#6669]
Using the latest version of Ghost 0.7.x I can easily add a header and footer on every page through the Admin page. When it comes to including Disqus style comments - this really has to be a manual edit of post.hbs and I could not find any way of injecting a footer on each post. 

When a newcomer starts using comments on their blog (a great way to get feedback, right?) they are probably going to need to edit post.hbs.

Example article on Docker on my blog:

http://blog.alexellis.io/dockerlabs/

I've tried searching for several terms in the issues page but didn't find any duplicate issue yet?

Blogs + comments are a really common combination.. is this something you guys would put on your roadmap?

**Comment by acburdine** [comment]
Hey @alexellis!

There's a wish list over at ideas.ghost.org where you can go an submit your idea. The github issue tracker is for issues with the software itself :smile:

Also, this seems very specific. Creating a full blown template editor just for one file seems a bit overkill. Also, there's the fact to consider that not everyone stores their themes on the same server as the blog, and not everyone has the same exact post.hbs layout.

That said though, feel free to submit it to the wishlist!

**Comment by alexellis** [comment]
@acburdine I think even your founder uses Disqus comments in this way.. head over to his blog entry on Moving to Singapore ... maybe he and I are the only people who need this? ;-) I'll raise on the ideas site, thanks for the link.

**Comment by acburdine** [comment]
I see what you mean now, apologies for that :smile: And I use Disqus too. However, I still think an injector just for posts would be far too specific. If it's just Disqus, I'd feel it's too limited of a use case. The code injection utilities for header and footer can be used for a wide array of things. But if you submit the idea and enough people vote on it then sure!

**Comment by alexellis** [comment]
@acburdine So now there are three of us in the world who use Disqus... but still it's not mainstream enough to warrant an easy way to install it on Ghost?

What do you... yourself do when you install Disqus? Do you also have to hack post.hbs and paste in a JS snippet after </footer> ? 

If so then that really is not a selling point for using Ghost... having to hack code to integrate comments on a blog with over a million users?

**Comment by acburdine** [comment]
I put the actual JS in using the code injection screen, and then I edit post.hbs and drop in a single html div I think. Disqus has a good how-to on their website.

And I think there's a decent amount of Ghost users who use Disqus. I just don't see why a quick, maybe 30 second edit of a theme warrants an entirely new feature. Plus there's the other section where you can add something for Disqus -> in the post loop section on the front page you can add something to show the current number of comments on that post.

I've heard that there are themes that come with these things already included, and the user just has to go into the code injection footer and enter a couple of Disqus specific variables. I know there's one of those themes for Casper somewhere.

**Comment by acburdine** [comment]
And I don't really hack post.hbs, I just edit it, which IMO isn't really hacking as themes are designed with easy editing in mind.

**Comment by ErisDS** [comment]
Hey @alexellis, the reason we have a wishlist (linked to from the Ghost admin and our issue template) rather than having feature requests here on GitHub, is to collect together ideas for things that would be useful and build traction around those ideas - to find out which problems 

When we had feature requests here on GitHub, we often found they got "stuck" and the reason for that is to do with how feature requests are often proposed - of which this is an example. The issue is asking specifically for a code injection section for post foot - but the problem you're having is you want an easy way to add Disqus... so why not ask for better Disqus support? By jumping and head and proposing a  solution it makes it harder to talk about the problem, understand it properly, and solve the problem well.

The underlying problem of wanting to make it easier to integrate comments with Ghost is a valid problem. There's a super generic item [here on the wishlist](http://ideas.ghost.org/forums/285309-wishlist/suggestions/7192519-comments) which has plenty of traction. We're experimenting with various things to do with apps and integrations at the moment & I'm quite certain the right answer will fall out of that work.

Adding an extra hook to the code injection screen isn't a good solution for the underlying problem of including Disqus because Disqus needs 2, sometimes 3 pieces of code injected: the script, the html for the location of the comments and if you want it, the additional stuff for comment counts. If we add one extra slot on the code injection screen, then it's a slippery slope to having an enormous list of them for various use cases.

TLDR; easier integration of 3rd party tools, including commenting tools like Disqus is a big problem that we're working on, but this isn't the right solution.

**Comment by lexrus** [comment]
Hi @alexellis 
You can try [my gist](https://gist.github.com/lexrus/ff6d91d29247956a81970a44427b0c64).
It's unnecessary to hack post.hbs anymore.
